src/main/java/org/opensearch/ad/transport/SearchAnomalyResultTransportAction.java

```diff
@@ -11,32 +11,163 @@
+            String resultIndexAggName = "result_index";
+            SearchSourceBuilder searchResultIndexBuilder = new SearchSourceBuilder();
+            AggregationBuilder aggregation = new TermsAggregationBuilder(resultIndexAggName)
+                .field(AnomalyDetector.RESULT_INDEX_FIELD)
+                .size(MAX_DETECTOR_UPPER_LIMIT);
+            searchResultIndexBuilder.aggregation(aggregation).size(0);
+            SearchRequest searchResultIndex = new SearchRequest(AnomalyDetector.ANOMALY_DETECTORS_INDEX).source(searchResultIndexBuilder);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                // Search result indices of all detectors. User may create index with same prefix of custom result index
+                // which not used for AD, so we should avoid searching extra indices which not used by anomaly detectors.
+                client.search(searchResultIndex, ActionListener.wrap(allResultIndicesResponse -> {
+                    Aggregations aggregations = allResultIndicesResponse.getAggregations();
+                    StringTerms resultIndicesAgg = aggregations.get(resultIndexAggName);
+                    List<StringTerms.Bucket> buckets = resultIndicesAgg.getBuckets();
+                    Set<String> resultIndicesOfDetector = new HashSet<>();
+                    if (buckets == null) {
+                        searchHandler.search(request, listener);
+                        return;
+                    }
+                    buckets.stream().forEach(b -> resultIndicesOfDetector.add(b.getKeyAsString()));
+                    List<String> targetIndices = new ArrayList<>();
+                    for (String index : customResultIndices) {
+                        if (resultIndicesOfDetector.contains(index)) {
+                            targetIndices.add(index);
+                        }
+                    }
+                    if (targetIndices.size() == 0) {
+                        // No custom result indices used by detectors, just search default result index
+                        searchHandler.search(request, listener);
+                        return;
+                    }
+                    MultiSearchRequest multiSearchRequest = new MultiSearchRequest();
+                    for (String index : targetIndices) {
+                        multiSearchRequest
+                            .add(new SearchRequest(index).source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()).size(0)));
+                    }
+                    List<String> readableIndices = new ArrayList<>();
+                    readableIndices.add(ALL_AD_RESULTS_INDEX_PATTERN);
+
+                    context.restore();
+                    // Send multiple search to check which index uer has permission to read. If search all indices directly,
+                    // search request will throw exception if user has no permission to search any index.
```

Comment:
I'm unsure why this is necessary. Why can't a single search over all indices work? And if there's some permissions error, then just propagate the error?

Reply:
I guess you mean we can send some search request like this `new SearchRequest(CUSTOM_RESULT_INDEX_PREFIX + "*")`. But this request will return security exception directly without telling us which index the user has no permission to read. From security plugin team, this is to avoid leaking security related information so user won't know which index they have permission and which they don't. 

Discussed with security plugin team, no easy way to solve this. So just use multiple search to check which index the user has no permission to read.

Reply:
sure, will remove this debug log